### PR TITLE
Enable local project analysis through command-line.

### LINKED ISSUE #204: command line option to execute pana against local directory

### PR DESCRIPTION
**Pull request by isoos**
- Creates backup of `pubspec.yaml` and restores it.
- Fixes #204

### CODE REVIEW

File: bin/pana.dart [hunk 1]
```diff
@@ -106,15 +99,24 @@ main(List<String> args) async {
   try {
     try {
       var analyzer = await PackageAnalyzer.create(pubCacheDir: tempPath);
-      var summary = await analyzer.inspectPackage(pkg,
-          version: version, pubHostedUrl: result['hosted-url']);
 
+      final target = result.rest.first;
+      final isLocalDir = await FileSystemEntity.isDirectory(target);
```
Comment (kevmoo) [inline bin/pana.dart]: Uh...not ideal. there is a package `test` for example.

I'd vote for `--source` with `site` (default) and `directory` as options.

Comment (isoos) [reply bin/pana.dart]: good point. I'm not sure why you would need `site`, I've added `--source <directory>`, PTAL

Comment (kevmoo) [reply bin/pana.dart]: Ah! See below for pub global activate

<package> is usually the package name (from hosted).

--source let's you say "no, <package> is a directory", etc.

Would be nice to align w/ what pub does here anyway...

```
Usage: pub global activate <package...>
-h, --help              Print this usage information.
-s, --source            The source used to find the package.
                        [git, hosted (default), path]

-f, --features          Feature(s) to enable.
-F, --omit-features     Feature(s) to disable.
    --no-executables    Do not put executables on PATH.
-x, --executable        Executable(s) to place on PATH.
    --overwrite         Overwrite executables from other packages with the same name.

Run "pub help" to see global options.
```

Comment (isoos) [reply bin/pana.dart]: How about now?

**Comment by isoos** [comment]
> Wondering if some sort of warning/FYI message would be good – telling folks that their pubspec.yaml file will be messed with?

In the log or upfront?

**Comment by kevmoo** [comment]
> In the log or upfront?

🤷‍♂️

I'd almost argue for a forced "are you sure?" prompt that can be overridden w/ --force

**Comment by isoos** [comment]
> I'd almost argue for a forced "are you sure?" prompt that can be overridden w/ --force

I think the `--source path` is a strong enough intent. I've updated the code using a `--no-warning` flag, and if you omit it, the analysis will be delayed by 15 seconds.

**Pull request merged**